torch/distributed/tensor/parallel/style.py

```diff
@@ -589,3 +591,102 @@ def __init__(
             if _prepare_output is not None
             else _get_prepare_output(output_layouts, use_local),
         )
+
+
+class PrepareModuleInput(ParallelStyle):
+    """
+    :class:`PrepareModuleInput` enables users to annotate :class:`torch.Tensor` or :class:`DTensor`
+    inputs with ``input_layouts`` and ``output_layouts`` so that each input can be converted to
+    :class:`DTensor` based on the annotation. Specifically, a DTensor will be created
+    from the input Tensor based on ``input_layouts`` and then redistributed to another
+    DTensor based on ``output_layouts``.
+
+    When the input is not a :class:`torch.Tensor` or :class:`DTensor`, if no layout is
+    specified, it will be a no-op. Otherwise, it will throw an error.
+    """
+
+    def __init__(
+        self,
+        input_layouts: LayoutsType = Shard(0),
+        output_layouts: LayoutsType = Replicate(),
+        use_local: bool = False,
+    ) -> None:
+        """
+        Args:
+            input_layouts (Union[Placement, Tuple[Placement, ...]]):
+                The layout of input tensor(s) which DTensor will be created upon.
+            output_layouts (Union[Placement, Tuple[Placement, ...]]):
+                The layout of input tensor(s) which created DTensor will be redistributed to.
+            use_local (bool):
+                Whether to convert the DTensor to local tensor.
+
+        Returns:
+            None.
+
+        Example::
+        >>> # xdoctest: +SKIP(failing)
+        >>> from torch.distributed.tensor.parallel import parallelize_module, PrepareModuleInput
+        >>> ...
+        >>> parallelize_plan = {
+        >>>     "attn": PrepareModuleInput(),   # The input of attn will be converted to Sharded DTensor.
```

Comment:
nit: input of attn will be converted to Sharded DTensor and redistributed to Replicated DTensor?